pandas/tseries/offsets.py

```diff
@@ -912,7 +919,9 @@ def apply(self, other):
     @apply_index_wraps
     def apply_index(self, i):
         shifted = liboffsets.shift_months(i.asi8, self.n, self._day_opt)
-        return i._shallow_copy(shifted)
+        # TODO: going through __new__ raises on call to _validate_frequency;
+        #  are we passing incorrect freq?
```

Comment:
Does this todo needs to be discussed / handled now?

Reply:
I'm happy thinking of this as a note to future-me.  Really looking forward to getting the current crop of PRs done with and DatetimeArray out the door.